src/key.h

```diff
@@ -170,6 +170,23 @@ struct CExtKey {
     bool Derive(CExtKey& out, unsigned int nChild) const;
     CExtPubKey Neuter() const;
     void SetMaster(const unsigned char* seed, unsigned int nSeedLen);
+    template <typename Stream>
```

Comment:
In the Encode and Decode functions above, we should use the BIP32_EXTKEY_SIZE constant, since the code below requires the proper array size.  Also we should change that constant to an enum since its defined in a header file.